amazon-redshift-plugin/src/main/java/io/cdap/plugin/amazon/redshift/RedshiftDBRecord.java

```diff
@@ -0,0 +1,123 @@
+   */
+  @SuppressWarnings("unused")
+  public RedshiftDBRecord() {
+  }
+
+  @Override
+  protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordBuilder, Schema.Field field,
+                             int columnIndex, int sqlType, int sqlPrecision, int sqlScale) throws SQLException {
+    ResultSetMetaData metadata = resultSet.getMetaData();
+    String columnTypeName = metadata.getColumnTypeName(columnIndex);
+    if (isUseSchema(metadata, columnIndex)) {
+      setFieldAccordingToSchema(resultSet, recordBuilder, field, columnIndex);
+    } else if (sqlType == Types.TIMESTAMP && columnTypeName.equalsIgnoreCase("timestamp")) {
+      Timestamp timestamp = resultSet.getTimestamp(columnIndex, DBUtils.PURE_GREGORIAN_CALENDAR);
+      if (timestamp != null) {
+        ZonedDateTime zonedDateTime = OffsetDateTime.of(timestamp.toLocalDateTime(), OffsetDateTime.now().getOffset())
+          .atZoneSameInstant(ZoneId.of("UTC"));
+        Schema nonNullableSchema = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable() : field.getSchema();
+        setZonedDateTimeBasedOnOuputSchema(recordBuilder, nonNullableSchema.getLogicalType(),
+                                           field.getName(), zonedDateTime);
+      } else {
+        recordBuilder.set(field.getName(), null);
+      }
+    } else if (sqlType == Types.TIMESTAMP && columnTypeName.equalsIgnoreCase("timestamptz")) {
+      OffsetDateTime timestamp = resultSet.getObject(columnIndex, OffsetDateTime.class);
+      if (timestamp != null) {
+        recordBuilder.setTimestamp(field.getName(), timestamp.atZoneSameInstant(ZoneId.of("UTC")));
+      } else {
+        recordBuilder.set(field.getName(), null);
+      }
+    } else {
+      int columnType = metadata.getColumnType(columnIndex);
+      if (columnType == Types.NUMERIC) {
+        Schema nonNullableSchema = field.getSchema().isNullable() ?
+          field.getSchema().getNonNullable() : field.getSchema();
+        int precision = metadata.getPrecision(columnIndex);
+        if (precision == 0 && Schema.Type.STRING.equals(nonNullableSchema.getType())) {
+          // When output schema is set to String for precision less numbers
+          recordBuilder.set(field.getName(), resultSet.getString(columnIndex));
+        } else if (Schema.LogicalType.DECIMAL.equals(nonNullableSchema.getLogicalType())) {
+          BigDecimal orgValue = resultSet.getBigDecimal(columnIndex);
+          if (orgValue != null) {
+            BigDecimal decimalValue = new BigDecimal(orgValue.toPlainString())
+              .setScale(nonNullableSchema.getScale(), RoundingMode.HALF_EVEN);
+            recordBuilder.setDecimal(field.getName(), decimalValue);
+          }
+        }
+        return;
+      }
+      setField(resultSet, recordBuilder, field, columnIndex, sqlType, sqlPrecision, sqlScale);
+    }
+  }
+
+  private void setZonedDateTimeBasedOnOuputSchema(StructuredRecord.Builder recordBuilder,
```

Comment:
Typo in `setZonedDateTimeBasedOnOuputSchema`

---
Suggestion
```suggestion
  private void setZonedDateTimeBasedOnOutputSchema(StructuredRecord.Builder recordBuilder,
```